bouncer/src/repo/core/handler/fileservice/repo_file_manager.cpp

```diff
@@ -349,18 +329,10 @@ bool FileManager::upsertFileRef(
 	const uint32_t                               &size,
 	const repo::core::model::RepoRef::Metadata   &metadata)
 {
-	std::string errMsg;
-	bool success = true;
-
 	auto refObj = makeRefNode(id, link, type, size, metadata);
 	std::string collectionName = collectionNamePrefix + "." + REPO_COLLECTION_EXT_REF;
-	success = dbHandler->upsertDocument(databaseName, collectionName, refObj, true, errMsg);
-	if (!success)
-	{
-		repoError << "Failed to add " << collectionName << " file ref: " << errMsg;;
-	}
-
-	return success;
+	getDbHandler()->upsertDocument(databaseName, collectionName, refObj, true);
```

Comment:
This will throw on any database errors.

If there is an exception inserting the ref node, that is a fatal error, so the exception is caught in the client and the whole process terminated.

We could catch the exception and delete the file we just wrote, but in that case there is still the question of what to do next: it may be half the scene has already been written, and if there is a database failure the import can't continue. It seemed best since we can't guarantee to clean up properly, to just let the process fail and the maintenance scripts handle it when they next run.

Reply:
I'm less worried about orphan data, but I'm more thinking this will make the logs more cryptic and hard for us to know what happened based on it when we need to diagnose an issue

Unless they've significantly improved their message details, the mongo error will only typically tell you the exact error (e.g. duplicated id) and we wouldn't really know what was the bouncer action that trigged it, or which collection we were working with etc.

We may be able to guess with the other logs surrounding it on the stage, but there are situations where we're doing a bulk of operations I think and it may be helpful to get more insights on error 🤔 

so even if it's just a catch just to log some details about what it was trying to do then throw it again it may be useful? (whether it's in here or at the caller)

Reply:
Makes sense! I will rework the RepoExceptions to hold more information, such as nested exceptions, and at each stage outwards catch and rethrow where there is important context so by the time it gets to the outer handler we have information from throughout the stack. I'll update the connection testing to do the same too.